tap/tcp_reader.go

```diff
@@ -15,35 +15,34 @@ import (
  * Implements io.Reader interface (Read)
  */
 type tcpReader struct {
-	ident         string
-	tcpID         *api.TcpID
-	isClosed      bool
-	isClient      bool
-	isOutgoing    bool
-	msgQueue      chan api.TcpReaderDataMsg // Channel of captured reassembled tcp payload
-	pastData      []byte
-	data          []byte
-	progress      *api.ReadProgress
-	captureTime   time.Time
-	parent        *tcpStream
-	packetsSeen   uint
-	emitter       api.Emitter
-	counterPair   *api.CounterPair
-	reqResMatcher api.RequestResponseMatcher
+	ident           string
+	tcpID           *api.TcpID
+	isClosed        bool
+	isClient        bool
+	isOutgoing      bool
+	msgQueue        chan api.TcpReaderDataMsg // Channel of captured reassembled tcp payload
+	msgBuffer       []api.TcpReaderDataMsg
+	msgBufferMaster []api.TcpReaderDataMsg
```

Comment:
What's the benefit of buffering the messages over buffering the bytes?

Reply:
> Changed the approach from buffering the bytes to buffering the TCP packet's (`api.TcpReaderDataMsg`) since the timestamp `captureTime` should be rewinded as well.

from https://github.com/up9inc/mizu/pull/1062#issuecomment-1126286372

Also it's cleaner but probably using slightly more memory.

Reply:
Got it. It does make more sense.